scripts/moderation_loop.py

```diff
@@ -0,0 +1,592 @@
+4. analyzes new pending flags
+5. auto-resolves high-confidence cases
+6. sends questions for ambiguous cases via DM
+7. exits (non-blocking)
+
+designed to run on a schedule (e.g., every 5 minutes via GitHub Actions).
+
+usage:
+    uv run scripts/moderation_loop.py
+    uv run scripts/moderation_loop.py --dry-run
+"""
+
+from __future__ import annotations
+
+import argparse
+import asyncio
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import httpx
+from atproto import AsyncClient, models
+from pydantic import BaseModel, Field
+from pydantic_ai import Agent
+from pydantic_ai.models.anthropic import AnthropicModel
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from rich.console import Console
+
+console = Console()
+
+
+# --- settings ---
+
+
+class LoopSettings(BaseSettings):
+    """settings for moderation loop."""
+
+    model_config = SettingsConfigDict(
+        env_file=Path(__file__).parent.parent / ".env",
+        case_sensitive=False,
+        extra="ignore",
+    )
+
+    # moderation service
+    moderation_service_url: str = Field(
+        default="https://moderation.plyr.fm",
+        validation_alias="MODERATION_SERVICE_URL",
+    )
+    moderation_auth_token: str = Field(
+        default="", validation_alias="MODERATION_AUTH_TOKEN"
+    )
+
+    # AI
+    anthropic_api_key: str = Field(default="", validation_alias="ANTHROPIC_API_KEY")
+    anthropic_model: str = Field(
+        default="claude-sonnet-4-20250514", validation_alias="ANTHROPIC_MODEL"
+    )
+
+    # bluesky bot for DMs
+    bot_handle: str = Field(default="", validation_alias="NOTIFY_BOT_HANDLE")
+    bot_password: str = Field(default="", validation_alias="NOTIFY_BOT_PASSWORD")
+    recipient_handle: str = Field(
+        default="", validation_alias="NOTIFY_RECIPIENT_HANDLE"
+    )
+
+
+# --- models ---
+
+
+class ModerationDecision(BaseModel):
+    """parsed decision from natural language reply."""
+
+    action: str = Field(
+        description="one of: approve_all, reject_all, approve_specific, reject_specific, need_clarification"
+    )
+    track_identifiers: list[str] = Field(
+        default_factory=list,
+        description="track titles, numbers, or URIs mentioned for specific actions",
+    )
+    reasoning: str = Field(description="extracted reasoning or notes from the reply")
+    confidence: float = Field(ge=0.0, le=1.0, description="confidence in parsing")
+
+
+class ConversationContext(BaseModel):
+    """context from recent DM conversation."""
+
+    has_pending_question: bool
+    question_text: str | None = None
+    question_uris: list[str] = Field(default_factory=list)  # URIs from the question
+    reply_text: str | None = None
+    decision: ModerationDecision | None = None
+
+
+class FlagSummary(BaseModel):
+    """summary of a flag for the question."""
+
+    uri: str
+    track_title: str | None
+    artist_handle: str | None
+    matches: list[str]  # "Title by Artist"
+    ai_reasoning: str
```

Comment:
These look a bit weak and uninspired. why use a comment instead of a field description?